packages/desktop-client/src/components/reports/ReportOptions.tsx

```diff
@@ -0,0 +1,38 @@
+const balanceTypeOptions = [
+  { description: 'Expense', format: 'totalDebts' },
+  { description: 'Income', format: 'totalAssets' },
+  { description: 'Net', format: 'totalTotals' },
+];
+
+const groupByOptions = [
+  { description: 'Category' },
+  { description: 'Group' },
+  { description: 'Payee' },
+  { description: 'Account' },
+  { description: 'Month' },
+  { description: 'Year' },
+];
+
+const dateRangeOptions = [
+  { description: '1 month', name: 1 },
+  { description: '3 months', name: 2 },
+  { description: '6 months', name: 5 },
+  { vdescription: '1 year', name: 11 },
+  { description: 'All time', name: 'allMonths' },
+];
+
+export const ReportOptions = {
+  groupBy: groupByOptions,
+  balanceType: balanceTypeOptions,
+  dateRange: dateRangeOptions,
+};
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏in a lot of places in the codebase you're doing `ReportOptions.balanceType.find`. This is a loop with the time complexity of O(n). We could improve this to O(1) quite easily.

```suggestion
export const ReportOptions = {
  groupBy: groupByOptions,
  balanceType: balanceTypeOptions,
  balanceTypeMap: new Map(balanceTypeOptions.map({ description, format }) => [description, format]),
  dateRange: dateRangeOptions,
};
```

Usage:
```
ReportOptions.balanceTypeMap.get('Expense')
```

or another approach would be..
```
export BALANCE_TYPE_EXPENSE = 'totalDebts';

const balanceTypeOptions = [
   { description: 'Expense', format: BALANCE_TYPE_EXPENSE },
   { description: 'Income', format: 'totalAssets' },
   { description: 'Net', format: 'totalTotals' },
 ];
```

Usage: `BALANCE_TYPE_EXPENSE`

--

Personally I prefer a simple constant, but up to you.